python/ray/tune/trial_runner.py

```diff
@@ -234,6 +249,14 @@ def checkpoint_exists(cls, directory):
             (fname.startswith("experiment_state") and fname.endswith(".json"))
             for fname in os.listdir(directory))
 
+    @classmethod
+    def searcher_checkpoint_exists(cls, directory):
```

Comment:
nit: some more concise / reusable alternatives (if you use this perhaps move to util.py)

```
import glob

def file_exists(cls, directory, file_pattern):
	path_pattern = glob.glob(os.path.join(directory, file_pattern)
	return len(glob.glob(path_pattern)) > 0

def find_newest_dated_path(cls, directory, file_pattern):
	path_pattern = glob.glob(os.path.join(directory, file_pattern)
	return max(glob.glob(path_pattern))

# and move the literals below to variables
file_exists(directory, "experiment_state_searcher_ckpt*.pkl")
file_exists(directory, "experiment_state*.json")
find_newest_file(directory, "experiment_state_searcher_ckpt*.pkl") 
find_newest_file(directory, "experiment_state*.json")
```

in any case it'd be good to dedup `*checkpoint_exists` + `_find_newest_*ckpt`, which should reduce testing slightly too

Reply:
Agree it would be nice to deduplicate code.